cloudsync/tasks.py

```diff
@@ -53,6 +56,45 @@ def callback(bytes_uploaded):
     )
 
 
+@shared_task(bind=True)
+def transcode_from_s3(self, video_id):  # pylint: disable=unused-argument
+    """
+    Given an S3 object key, transcode that object using a video pipeline, overwrite the original when done.
+    :param self: The celery task
+    :param key: The S3 video object key
+    :return: None
+    """
+    video_file = VideoFile.objects.get(video__id=video_id, encoding='original')
+    video = video_file.video
+
+    video_input = {
+        'Key': video_file.s3_object_key,
+    }
+
+    # Generate an output video file for each encoding (assumed to be HLS)
+    outputs = [{
+        'Key': video.transcode_key(preset),
+        'PresetId': preset,
+        'SegmentDuration': '10.0'
+    } for preset in settings.ET_PRESET_IDS]
+
+    playlists = [{'Format': 'HLSv4',
+                  'Name': video.transcode_key('_index'),
+                  'OutputKeys': [output['Key'] for output in outputs]}]
+
+    # Generate thumbnails for the 1st encoding (no point in doing so for each).
+    basename, _ = os.path.splitext(video_file.s3_object_key)
+    outputs[0]['ThumbnailPattern'] = "thumbnails/{}_thumbnail_{{count}}".format(basename)
```

Comment:
Should probably validate the length of `settings.ET_PRESET_IDS` and raise an appropriate error at the start of the method since you're depending on it to be nonzero here.